app/models/offering.rb

```diff
@@ -0,0 +1,13 @@
+class Offering < ApplicationRecord
+  has_one :loan
+  has_many :investments
+  has_many :invested_investments, -> { where status: 'invested' },
+           class_name: 'Investment',
+           foreign_key: 'offering_id'
+  has_many :processed_transactions, -> { where status: 'processed' }
+  has_many :transactions
+
+  def funds_available
+    amount - invested_investments.sum(:amount)
```

Comment:
invested_investments is a database call.  does this cause n + 1?  hint: keep an eye on your ruby console when you test this stuff.

Reply:
How would you recommend?

Instead, I could create a virtual attribute for `sum_of_invested_investments` and put the database call in there? But I wouldn't expect that to change performance.